### ISSUE TITLE
Style/ConstantName gives a false positive for SCREAMING_SNAKE_CASE with accents

### ISSUE THREAD

**Issue by timrogers** [bbatsov/rubocop#5016]
`Rubocop::Cop::Style::ConstantName` enforces that constant names must be written using SCREAMING_SNAKE_CASE, by comparing them to its regex, `SNAKE_CASE`.

Capital letters can have accents, but the regular expression only permits unaccented letters A-Z. We should also allow capital letters with accents.

--------

## Expected behavior

```ruby
MATÍAS = "matías"
```

should be permitted by the cop, just as:

```ruby
MATIAS = "matías"
```

is permitted.

## Actual behavior

Rubocop detects an offence.

## Steps to reproduce the problem

Define a constant using what looks like SCREAMING_SNAKE_CASE, but including capital letterswith accents.

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-darwin16)
```
